pkg/chunkenc/gzip.go

```diff
@@ -309,9 +315,22 @@ func (c *MemChunk) Size() int {
 	return ne
 }
 
+// Blocks implements Chunk.
+func (c *MemChunk) Blocks() int {
+	return len(c.blocks)
+}
+
 // SpaceFor implements Chunk.
-func (c *MemChunk) SpaceFor(*logproto.Entry) bool {
-	return len(c.blocks) < blocksPerChunk
+func (c *MemChunk) SpaceFor(e *logproto.Entry) bool {
+	if c.targetSize > 0 {
+		// This is looking to see if the uncompressed lines will fit which is not
+		// a great check, but it will guarantee we are always under the target size
+		newHBSize := c.head.size + len(e.Line)
+		return (c.cutBlockSize + newHBSize) < c.targetSize
+	} else {
```

Comment:
the else is not necessary

Reply:
yeah, our linter agrees with you but personally I prefer it this way... but I guess it's me vs you and the linter and I'm gonna lose this one :)

Reply:
You are !